packages/codecs/src/known/babe.ts

```diff
@@ -0,0 +1,157 @@
+import * as $ from '@delightfuldot/shape';
+import { $AccountId32, $Header } from '../generic';
+
+/*
+ * The weight of an authority.
+ */
+export const $BabeAuthorityWeight = $.u64;
+
+/*
+ * Randomness type required by BABE operations.
+ */
+export const $Randomness = $.FixedHex(32);
+
+/*
+ * Types of allowed slots.
+ */
+export const $AllowedSlots = $.FlatEnum([
+  // Only allow primary slots.
+  'PrimarySlots',
+  // Allow primary and secondary plain slots.
+  'PrimaryAndSecondaryPlainSlots',
+  // Allow primary and secondary VRF slots.
+  'PrimaryAndSecondaryVRFSlots',
+]);
```

Comment:
Let's define corresponding types for each defined codecs in this package. These 3 are just example, please check other places as well.